src/search/tasks/root_task.cc

```diff
@@ -266,18 +285,24 @@ static vector<vector<set<FactPair>>> read_mutexes(istream &in, const vector<Expl
       aware of.
     */
     for (int i = 0; i < num_mutex_groups; ++i) {
-        check_magic(in, "begin_mutex_group");
-        int num_facts;
-        in >> num_facts;
+        file_parser.read_magic_line("begin_mutex_group");
+        int num_facts = file_parser.read_line_int("number of facts in mutex group");
+        if (num_facts < 1) {
+            file_parser.error("Number of facts in mutex group is less than 1, should be at least 1.");
+        }
         vector<FactPair> invariant_group;
         invariant_group.reserve(num_facts);
         for (int j = 0; j < num_facts; ++j) {
```

Comment:
Don't we already have a function that reads a list of facts? And doesn't it also need the check that there are no duplicate facts?